src/identity/mnemonic.ts

```diff
@@ -158,6 +177,50 @@ export class Mnemonic extends MasterPublicKey implements IdentityInterface {
     return pset.toBase64();
   }
 
+  async signPsetV2(psetBase64: string): Promise<string> {
+    if (!this.ecclib) throw new Error('ecclib is missing, cannot sign pset');
+
+    const pset = Pset.fromBase64(psetBase64);
+    const signer = new Signer(pset);
+
+    pset.inputs.forEach((input, i) => {
```

Comment:
I would use `for ... of ...` loop instead of `forEach`. The drawback of forEach is, in case of error, you have an anonymous function in traceback so harder to debug